app/actions/app/server.test.ts

```diff
@@ -0,0 +1,213 @@
+jest.mock('@managers/websocket_manager');
+jest.mock('@utils/log');
+jest.mock('@utils/push_proxy');
+jest.mock('@utils/server');
+jest.mock('@actions/remote/general');
+jest.mock('@actions/remote/systems');
+jest.mock('react-native-navigation');
+
+const translations = getTranslations(DEFAULT_LOCALE);
+const intl = createIntl({locale: DEFAULT_LOCALE, messages: translations});
+const theme = Preferences.THEMES.denim;
+
+describe('initializeSecurityManager', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should return when no servers are found', async () => {
+        const mockQuery = {
+            fetch: jest.fn().mockResolvedValueOnce([]),
+        } as unknown as Query<ServersModel>;
+        jest.mocked(queryAllActiveServers).mockReturnValueOnce(mockQuery);
+        await Actions.initializeSecurityManager();
+        expect(SecurityManager.init).not.toHaveBeenCalled();
+    });
+
+    it('should initialize SecurityManager with querying configurations', async () => {
+        const servers = [{url: 'server1'}, {url: 'server2'}];
+        const mockQuery = {
+            fetch: jest.fn().mockResolvedValueOnce(servers),
+        } as unknown as Query<ServersModel>;
+        jest.mocked(queryAllActiveServers).mockReturnValueOnce(mockQuery);
+        jest.mocked(DatabaseManager.getServerDatabaseAndOperator).mockImplementation((serverUrl) => ({database: `db_${serverUrl}`} as unknown as ServerDatabase));
+        const config = {
+            AboutLink: '',
+            MobileEnableBiometrics: 'true',
+            MobilePreventScreenCapture: 'true',
+            MobileJailbreakProtection: 'true',
+        } as unknown as ClientConfig;
+        jest.mocked(getConfig).mockImplementation((db) => Promise.resolve({
+            ...config,
+            key: `config_${db}`,
+        }));
+
+        await Actions.initializeSecurityManager();
+
+        expect(SecurityManager.init).toHaveBeenCalledWith({
+            server1: {...config, key: 'config_db_server1'},
+            server2: {...config, key: 'config_db_server2'},
+        }, undefined);
+    });
+
+    it('should log error when querying configuration fails', async () => {
+        const servers = [{url: 'server1'}];
+        const mockQuery = {
+            fetch: jest.fn().mockResolvedValueOnce(servers),
+        } as unknown as Query<ServersModel>;
+        jest.mocked(queryAllActiveServers).mockReturnValueOnce(mockQuery);
+        jest.mocked(DatabaseManager.getServerDatabaseAndOperator).mockImplementation(() => {
+            throw new Error('test error');
+        });
+
+        await Actions.initializeSecurityManager();
+
+        expect(logError).toHaveBeenCalledWith('initializeSecurityManager', expect.any(Error));
+    });
+});
+
+// Tests for switchToServer
+describe('switchToServer', () => {
+    it('should log error when server is not found', async () => {
+        (getServer as jest.Mock).mockResolvedValueOnce(null);
```

Comment:
Prefer `jest.mocked` over `as jest.Mock` (you have more places like this in this file)